contracts/USM.sol

```diff
@@ -114,20 +98,15 @@ contract USM is ERC20 {
      */
     function _usmToEth(uint _usmAmount) internal view returns (uint) {
         require(_usmAmount > 0, "USM Amount must be more than 0");
-        (uint usdPrice, uint decimalShift) = _oracleData();
-        return (_usmAmount.mul(10**decimalShift)).div(usdPrice);
+        return divd(_usmAmount, _oraclePrice());
     }
 
     /**
-     * @notice Retrieve the latest price and decimal shift of the
-     * price oracle.
+     * @notice Retrieve the latest price of the price oracle.
      *
-     * @return (latest price, decimal shift)
+     * @return price in UNIT
      */
-    function _oracleData() internal view returns (uint, uint) {
-        return (
-            IOracle(oracle).latestPrice(),
-            IOracle(oracle).decimalShift()
-        );
+    function _oraclePrice() internal view returns (uint) {
```

Comment:
We can make the oracle return a price in a UNIT format directly usable by the rest of the contract.